src/components/media/MediaModal.jsx

```diff
@@ -6,50 +6,81 @@ import mediaStyles from '../../styles/isomer-cms/pages/Media.module.scss';
 import elementStyles from '../../styles/isomer-cms/Elements.module.scss';
 import MediaCard from './MediaCard';
 import MediaUploadCard from './MediaUploadCard';
+import { MediaSearchBar } from './MediaSearchBar';
 import LoadingButton from '../LoadingButton';
 
-export default class MediasModal extends Component {
+export default class MediaModal extends Component {
   constructor(props) {
     super(props);
     this.state = {
       medias: '',
+      filteredMedias: '',
       selectedFile: null,
     };
   }
 
   async componentDidMount() {
-    const { siteName, type } = this.props;
+    const { siteName, type, imageSearchTerm } = this.props;
     const mediaRoute = type === 'file' ? 'documents' : 'images';
     try {
       const { data: { documents, images } } = await axios.get(`${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}/${mediaRoute}`, {
         withCredentials: true,
       });
       if (_.isEmpty(documents || images)) {
-        this.setState({ medias: [] });
+        this.setState({ medias: [], filteredMedias: [], searchTerm: imageSearchTerm });
       } else {
-        this.setState({ medias: documents || images });
+        let filteredMedias
+        if (type === 'file') {
+          filteredMedias = this.filterMediaByFileName(documents, imageSearchTerm)
+        } else {
+          filteredMedias = this.filterMediaByFileName(images, imageSearchTerm)
+        }
+
+        this.setState({ medias: documents || images, filteredMedias, searchTerm: imageSearchTerm });
       }
     } catch (err) {
       console.error(err);
     }
   }
 
+  filterMediaByFileName = (medias, filterTerm) => {
+    const filteredMedias = medias.filter((media) => {
+      if (media.fileName.toLowerCase().includes(filterTerm.toLowerCase())) return true
```

Comment:
cleaning, unnecessary `if` check here, can simplify lines 48, 49 to:
`return media.fileName.toLowerCase().includes(filterTerm.toLowerCase())`